src/vstest.console/CommandLine/TestCaseFilter.cs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestPlatform.Common.Filtering;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    
+    /// <summary>
+    /// Unit test filtering
+    /// </summary>
+    public class TestCaseFilter : ITestCaseFilter
+    {
+        public TestCaseFilter()
+        {
+            
+        }
+
+        public void Initialize(string filterString)
+        {
+            ValidateFilter(filterString);
+        }
+
+        /// <summary>
+        /// Filter tests
+        /// </summary>
+        public IEnumerable<TestCase> FilterTests(IEnumerable<TestCase> testCases)
+        {
+            EqtTrace.Verbose("TestCaseFilter.FilterTests : Test Filtering invoked.");
+
+            List<TestCase> filteredList;
+
+            try
+            {
+                filteredList = GetFilteredTestCases(testCases);
+            }
+            catch (Exception ex)
+            {
+                EqtTrace.Error("TestCaseFilter.FilterTests : Exception during filtering : {0}", ex.ToString());
+                throw;
+            }
+
+            return filteredList;
+        }
+
+        private void ValidateFilter(string filterString)
+        {
+            if (string.IsNullOrEmpty(filterString))
+            {
+                _filterExpression = null;
```

Comment:
>_filterExpression [](start = 16, length = 17)

Nit: please use this for member variables and remove '_'.

Reply:
Removed '_'. filterExpression is a static type. Cannot access it with an instance (which is this in our case)